app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt

```diff
@@ -214,6 +214,7 @@ fun ConversationScreen(
         conversationMessages = conversationMessagesViewModel.infoMessage,
         conversationMessagesViewModel = conversationMessagesViewModel,
         onPingClicked = messageComposerViewModel::sendPing,
+        onSelfDeletingMessageRead = messageComposerViewModel::startSelfDeletion,
```

Comment:
I think it makes more sense to keep it the way it is, for now we are calling onSelfDeletingMessageRead when we see message for the first, the same goes for marking a message read so it would make sense here, but it could change in the future also for asset we only start self deletion when the user downloads the asset, so already here we would have problems

Reply:
That's fair.

But in any case, the composable knows a bit too much.

It could be `onMessageContententRead` or similar. So it can be a generic lambda used to inform the ViewModel that some message and all of its content (included displaying of images) is _completely_ visible.

Or, the same lambda could take a parameter.

Like:

```
enum class ContentAvailability { 
    /**
     * Content is only half-available. For example, the asset message
     * is visible to the user, but the asset wasn't downloaded yet.
     */
    PARTIAL,

    /**
     * Content is only fully-available to the user. For example, the asset
     * message is visible and downloaded, so the user can see it.
     */
    COMPLETE
}

// For text messages
onMessageContentRead(COMPLETE)

// For assets, check if it's fully visible
onMessageContentRead(if(isAssetDownloaded) COMPLETE else PARTIAL)
```

This way:
- The view worries about displaying and informing the `ViewModel`
- The `ViewModel` can figure out what needs to be done for each kind of message (_e.g._ marking as read, and/or triggering self deletion, etc)

I'm happy for now, but I'd prefer to keep the view as dumb as possible regarding business logic and reduce the number of lambdas to make a simpler interface between V-VM

Reply:
This class is a mess on it's own we could take your proposition into the refactoring because there are more places where we could extract the logic to viewmodel i feel like this needs a seperate PR with a seperate place for discussion